File: AEDSII-TP/AEDII-master/tps/entrada e saida/tp01/TP01Q02 - Palíndromo em C/palindromo.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

/*
 * verifica se a cadeia de caracteres é um palindromo
 * @param char *s - cadeia de caracteres a ser verificada
 * @return bool resp - retorna true se palindromo e false caso contrario
 */
bool teste (char *s){
	
	bool resp = true;
	int a = strlen(s) - 1;
	int x = 0;

	for(; x < a ; x++){
		if(s[x] != s[a]){
			resp = false;
			x = a;
		}
		a--;
	}

	return resp;
}


int main (void){

	char *s = (char *) calloc (1001 , sizeof (char));
	
	s[1000] = '\0';
	fgets (s , 999 , stdin);
	s[strlen(s) - 1] = '\0';

	while(strcmp( s , "FIM") != 0){

		if(teste(s)){
			printf("SIM\n");
		}
		else{
			printf("NAO\n");
		}
		
		fgets(s , 999, stdin);
		s[strlen(s) - 1] = '\0';

	}
}

File: AEDSII-TP/tp1/tp01/TP01Q01 - Palíndromo em Java/palindromo.java
/*
Nome: Diego Da Silva Ferrera
Matricula: 71 06 78
*/

public class palindromo {

/*
 *testa se a String é um palindromo
 *@param String s - string a ser verificada
 *@return bollean resp - retorna true caso a string seja um palindromo e false caso contrario
 * */
public static boolean teste (String s){

	boolean resp = true;
	int a = s.length() - 1;

	for(int x = 0; x < a; x++){
		if(s.charAt(x) != s.charAt(a)){
			resp = false;
			x = a;
		}
		a--;
	}
	return resp;

}

public static void main(String[] args){
	
	boolean x = true;
	String s = MyIO.readLine();

	//metodo para verificar se string é igual a "FIM"
	while(!(s.length() == 3 && 
		s.charAt(0) == 'F' && 
		s.charAt(1) == 'I' && 
		s.charAt(2) == 'M')){

		
			if(teste(s)){
				System.out.println("SIM");
			}
			else{
				System.out.println("NAO");
			}	
		
		s = MyIO.readLine();

	}

}
}

